p2p/src/simulated/network.rs

```diff
@@ -632,6 +647,29 @@ impl<P: PublicKey> Sender<P> {
             processor,
         )
     }
+
+    /// Split this sender into two logical replicas that can route per message.
+    ///
+    /// The shared `router` receives the replica identifier, the requested recipients, and the
+    /// message payload, and returns the actual recipients to send to.
+    pub fn split_with<F>(self, router: F) -> (RoutedSender<P>, RoutedSender<P>)
```

Comment:
This approach allows us to [1] avoid throwing off bandwidth tracking and [2] avoid piping through the "split sender" all the way through the networking core.